tests/webview/ui/test_browse.py

```diff
@@ -65,20 +68,291 @@ def test_subject_categories_have_page_and_book_counts(base_url, selenium):
+
+    p1 = pagination[1]
+    assert not p1.is_disabled
+    assert p1.is_active
+
+    p2 = pagination[2]
+    assert not p2.is_disabled
+    assert not p2.is_active
+
+    next = pagination[-1]
+    assert not next.is_disabled
+    assert not next.is_active
+
+
+@markers.webview
+@markers.nondestructive
+def test_search_click_pagination(base_url, selenium):
+    # GIVEN the search results page
+    home = Home(selenium, base_url).open()
+    browse = home.header.click_search()
+    # Needs to contain a rare word, otherwise we may timeout when searching
+    # Postgres removes stop words automatically, so don't include those either
+    query = 'Amazing Aardvark Concepts'
+    search_results = browse.search(query)
+
+    # WHEN we click on the "next" pagination link
+    next = search_results.pagination[-1]
+    search_results = next.click_link()
+
+    # THEN the pagination is updated
+    assert type(search_results) is SearchResults
+
+    pagination = search_results.pagination
+    assert len(pagination) >= 4
+
+    prev = pagination[0]
+    assert not prev.is_disabled
+    assert not prev.is_active
+
+    p1 = pagination[1]
+    assert not p1.is_disabled
+    assert not p1.is_active
+
+    p2 = pagination[2]
+    assert not p2.is_disabled
+    assert p2.is_active
+
+    next = pagination[-1]
+    # assert next.is_disabled - will change if we have more than 2 pages of results in the future
+    assert not next.is_active
+
+
+@markers.webview
+@markers.nondestructive
+def test_advanced_search(base_url, selenium):
+    # GIVEN the advanced search page
+    home = Home(selenium, base_url).open()
+    browse = home.header.click_search()
+    advanced_search = browse.click_advanced_search_link()
+
+    # WHEN we select some filters and click submit
+    # The advanced search page sometimes replaces the search fields with new ones and erases
+    # whatever we typed, causing a StaleElementReferenceException. In this case, we retry.
+    max_tries = 2
+    for i in range(max_tries):
+        try:
+            search_results = advanced_search.fill_in_author('OpenStax') \
+                                            .fill_in_title('Concepts of Biology') \
+                                            .select_subject('Science and Technology') \
+                                            .fill_in_keywords('Amazing Aardvark') \
+                                            .select_type('book') \
+                                            .select_language('en') \
+                                            .select_publication_date(2018) \
+                                            .select_sort_by('pubDate') \
+                                            .submit()
+            break
+        except StaleElementReferenceException:
+            if i >= max_tries - 1:
+                raise
+
+    # THEN search results are displayed with the chosen filters
+    assert type(search_results) is SearchResults
+
+    expected_breadcrumbs = {
```

Comment:
This should be in the Given section of the test or a fixture. Any data that is used to "setup" the test, configure the environment, or provide data should be a fixture, preferably. GIVEN is composed of these setup items.

Reply:
also @pytest.mark.parametrize() might be a good idea. this would allow multiple search terms and maybe make the bookmarks based on those?